src/rpc/handlers/AMMInfo.h

```diff
@@ -0,0 +1,95 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include <data/BackendInterface.h>
+#include <rpc/BookChangesHelper.h>
+#include <rpc/RPCHelpers.h>
+#include <rpc/common/Types.h>
+#include <rpc/common/Validators.h>
+
+namespace rpc {
+
+/**
+ * @brief AMMInfoHandler returns information about AMM pools.
+ *
+ * This API is not documented in the rippled API documentation.
+ */
+class AMMInfoHandler {
+    std::shared_ptr<BackendInterface> sharedPtrBackend_;
+
+public:
+    struct Output {
+        // todo: use better type than json types
+        boost::json::value amount1;
+        boost::json::value amount2;
+        boost::json::value lpToken;
+        boost::json::array voteSlots;
+        boost::json::value auctionSlot = nullptr;
```

Comment:
no need nullptr, you can use auctionSlot.is_null() ,default constructed represents a null JSON element

Reply:
Fixed, thanks!